src/graphics.cpp

```diff
@@ -227,6 +227,11 @@ void Graphics::Freeze() {
 }
 
 void Graphics::Transition(TransitionType type, int duration, bool erase) {
+	if (erase == screen_erased) {
+		// Don't allow another erase or non-erase
+		return;
+	}
+
```

Comment:
This doesn't reproduce the behaviour of RPG_RT when using Show Screen in a non-erased screen. For example, try using Show Screen with the transition type Scroll Right.